docs/references/architecture/adr-114-undo-109.md

```diff
@@ -0,0 +1,168 @@
+| autofile                   | keep private       | 0                                      | 10             | low        | [autofile-url]    |
+| async                      | keep private       | 0                                      | 8              | low        | [async-url]       |
+| flowrate                   | keep private       | 0                                      | 10             | low        | [flowrate-url]    |
+| bits                       | keep private       | 0                                      | 26             | low        | [bits-url]        |
+| blocksync                  | keep private       | 0                                      | 6              | low        | [blocksync-url]   |
+| clist                      | keep private       | 0                                      | 24             | low        | [clist-url]       |
+| net                        | keep private       | 1                                      | 40             | low        | [net-url]         |
+| statesync                  | **🧹 make public** | 1                                      | 16             | medium     | [statesync-url]   |
+| evidence                   | **🧹 make public** | 1                                      | 26             | high       | [evidence-url]    |
+| consensus                  | **🧹 make public** | 1                                      | 64             | high       | [consensus-url]   |
+| indexer                    | **🧹 make public** | 2                                      | 100            | medium     | [indexer-url]     |
+| protoio                    | **🧹 make public** | 3                                      | 44             | low        | [protoio-url]     |
+| sync                       | **🧹 make public** | 3                                      | 172            | low        | [sync-url]        |
+| tempfile                   | keep private       | 4                                      | 16             | low        | [tempfile-url]    |
+| strings                    | keep private       | 4                                      | 14             | low        | [strings-url]     |
+| service                    | **🧹 make public** | 6                                      | 156            | low        | [service-url]     |
+| os                         | keep private       | 7                                      | 262            | low        | [os-url]          |
+| rand                       | keep private       | 7                                      | 317            | low        | [rand-url]        |
+| pubsub                     | **🧹 make public** | 7                                      | 169            | medium     | [pubsub-url]      |
+
+### Summary
+
+To summarize, these modules will remain public in v1 and marked as deprecated:
+- `evidence`
+- `consensus`
+- `indexer`
```

Comment:
why is indexer deprecated? the interface in the package should continue being public otherwise why have an interface? I think the implementations should be private but you should allow people to implement custom indexers, unless this system is being replaced?

Reply:
Last year, we decided that, moving forward (starting from `v1.x`), the data companion ([ADR 101](https://github.com/cometbft/cometbft/blob/main/docs/references/architecture/adr-101-data-companion-pull-api.md)) will progressively replace the indexer interface. ADR 101 is a better solution for implementing your own indexer.

Reply:
makes sense thank you